ledger.go

```diff
@@ -200,29 +210,122 @@ func (l *Ledger) Addresses(hrp string, addresses int) ([]*Address, error) {
 // SignHash attempts to sign the [hash] with the provided path [addresses].
 // [addresses] are appened to the [pathPrefix] (m/44'/9000'/0'/0).
 func (l *Ledger) SignHash(hash []byte, addresses []uint32) ([][]byte, error) {
-	msgHash := []byte{
-		CLA,
-		INSSignHash,
-		0x0,
-		0x0,
-	}
 	pathBytes, err := bip32bytes(pathPrefix, 3)
 	if err != nil {
 		return nil, err
 	}
 	data := []byte{byte(len(addresses))}
 	data = append(data, hash...)
 	data = append(data, pathBytes...)
-	msgHash = append(msgHash, byte(len(data)))
-	msgHash = append(msgHash, data...)
+	data = append([]byte{byte(len(data))}, data...)
 	fmt.Printf("signing hash: %X\n", hash)
-	resp, err := l.device.Exchange(msgHash)
+	resp, err := l.SendToLedger(CLA, INSSignHash, 0x00, 0x00, data)
 	if err != nil {
 		return nil, err
 	}
 	if !bytes.Equal(resp, hash) {
 		return nil, fmt.Errorf("returned hash %x does not match requested %x", resp, hash)
 	}
 
-	return l.collectSignatures(addresses)
+	return l.collectSignatures(addresses, INSSignHash, 0x01, 0x81)
+}
+
+func (l *Ledger) PrepareChunks(txn []byte, addresses []uint32, changePath []uint32) ([][]byte, error) {
+	if txn == nil || addresses == nil {
+		return nil, fmt.Errorf("chunks structure is undefined")
+	}
+	var chunks [][]byte
+	var preamble []byte
+	chunkNum := 0
+
+	pathBytes, err := bip32bytes(pathPrefix, 3)
+	if err != nil {
+		return nil, err
+	}
+	preamble = append(preamble, byte(len(addresses)))
+	preamble = append(preamble, pathBytes...)
```

Comment:
Is this actually correct? Should we be passing `addresses` into `bip32bytes`? This function is currently only using the length of `addresses`

Reply:
If you take a look at line 340 in handle_apdu_sign_transaction, that is why the number of addresses or rather requested sigs is required https://github.com/obsidiansystems/ledger-app-avalanche/blob/develop/src/apdu_sign.c